test_runner/src/main/kotlin/ftl/gc/android/CreateAndroidInstrumentationTest.kt

```diff
@@ -46,13 +48,24 @@ internal fun AndroidInstrumentationTest.setupTestTargets(
                     }
                     uniformSharding = UniformSharding().setNumShards(safeNumUniformShards)
                 } else {
-                    manualSharding = ManualSharding().setTestTargetsForShard(
-                        testShards.map {
-                            TestTargetsForShard().setTestTargets(it)
-                        }
-                    )
+                    manualSharding = createManualSharding(testFlagForShard, testShards)
                 }
             }
         }
     }
 }
+
+private fun createManualSharding(testFlagForShard: List<String>?, testShards: ShardChunks) =
+    if (testFlagForShard != null && testFlagForShard.isNotEmpty()) {
+        ManualSharding().setTestTargetsForShard(
+            testFlagForShard.map {
+                TestTargetsForShard().setTestTargets(it.split(',', ';'))
```

Comment:
IMO this split should be done earlier, probably inside `createAndroidArgs` function. Here we want to operate on well prepared and understandable domain structures, not encoded strings.

Reply:
Please have a look at the most recent changes about this 👍